functions/Export-DbaInstance.ps1

```diff
@@ -186,254 +212,198 @@ function Export-DbaInstance {
-                Get-ChildItem -ErrorAction Ignore -Path "$exportPath\$fileCounter-dbmail.sql"
-                if (-not (Test-Path "$exportPath\$fileCounter-dbmail.sql")) {
-                    $fileCounter--
-                }
+                # The first invocation to Export-DbaScript needs to have -Append:$false so that the previous file contents are discarded. Otherwise, the file would end up with duplicate SQL.
+                # The subsequent calls to Export-DbaScript need to have -Append:$true because this is a multi-step export and the objects are written to the same file.
+                $null = Get-DbaDbMailConfig -SqlInstance $server | Export-DbaScript -FilePath "$exportPath\dbmail.sql" -Append:$false -BatchSeparator $BatchSeparator -ScriptingOptionsObject $ScriptingOption -NoPrefix:$NoPrefix
+                $null = Get-DbaDbMailAccount -SqlInstance $server | Export-DbaScript -FilePath "$exportPath\dbmail.sql" -Append:$true -BatchSeparator $BatchSeparator -ScriptingOptionsObject $ScriptingOption -NoPrefix:$NoPrefix
+                $null = Get-DbaDbMailProfile -SqlInstance $server | Export-DbaScript -FilePath "$exportPath\dbmail.sql" -Append:$true -BatchSeparator $BatchSeparator -ScriptingOptionsObject $ScriptingOption -NoPrefix:$NoPrefix
+                $null = Get-DbaDbMailServer -SqlInstance $server | Export-DbaScript -FilePath "$exportPath\dbmail.sql" -Append:$true -BatchSeparator $BatchSeparator -ScriptingOptionsObject $ScriptingOption -NoPrefix:$NoPrefix
+
+                Get-ChildItem -ErrorAction Ignore -Path "$exportPath\dbmail.sql"
             }
 
             if ($Exclude -notcontains 'CentralManagementServer') {
-                $fileCounter++
                 Write-Message -Level Verbose -Message "Exporting Central Management Server"
                 Write-ProgressHelper -StepNumber ($stepCounter++) -Message "Exporting Central Management Server"
-                $null = Get-DbaRegServerGroup -SqlInstance $server | Export-DbaScript -FilePath "$exportPath\$fileCounter-regserver.sql" -Append:$Append -BatchSeparator 'GO' -NoPrefix:$NoPrefix
-                $null = Get-DbaRegServer -SqlInstance $server | Export-DbaScript -FilePath "$exportPath\$fileCounter-regserver.sql" -Append:$Append -BatchSeparator 'GO' -NoPrefix:$NoPrefix
-                Get-ChildItem -ErrorAction Ignore -Path "$exportPath\$fileCounter-regserver.sql"
-                if (-not (Test-Path "$exportPath\$fileCounter-regserver.sql")) {
-                    $fileCounter--
+                $outputFilePath = "$exportPath\regserver.xml"
+                if ($Overwrite) {
+                    $null = Export-DbaRegServer -SqlInstance $server -FilePath $outputFilePath -Overwrite
+                } else {
+                    $null = Export-DbaRegServer -SqlInstance $server -FilePath $outputFilePath
```

Comment:
This can be changed to one-line call:

`$null = Export-DbaRegServer -SqlInstance $server -FilePath $outputFilePath -Overwrite:$Overwrite`